manuskript/ui/editors/editorWidget.py

```diff
@@ -344,12 +352,33 @@ def modelDataChanged(self, topLeft, bottomRight):
         if topLeft.row() <= self.currentIndex.row() <= bottomRight.row():
             self.updateStatusBar()
 
-    #def rowsAboutToBeRemoved(self, parent, first, last):
-        #if self.currentIndex:
-            #if self.currentIndex.parent() == parent and \
-                                    #first <= self.currentIndex.row() <= last:
-                ## Item deleted, close tab
-                #self.mw.mainEditor.tab.removeTab(self.mw.mainEditor.tab.indexOf(self))
+    def rowsInserted(self, parent, first, last):
+        if self.currentIndex:
+            self.updateIndexFromID(fallback=self.currentIndex.parent())
+
+    def rowsAboutToBeRemoved(self, parent, first, last):
+        if not self.currentIndex:
+            return
```

Comment:
What's the use case where currentIndex is None? That doesn't happen, even a 'invalid index' is set in setCurrentModelIndex with `           self.currentIndex = QModelIndex()`
Same for rowsInserted

Reply:
Defensive coding against inconsistent state. That check was in the original code for one of the handlers and it made sense to me to have it so I made sure the signal handlers were equal in their check.

Which does not mean the check made sense to begin with if the base state is a QModelIndex.

Question though: the index paramter for setCurrentModelIndex() is by default set to None. I am not quite sure how its check if 'index.isValid()' could work out? I believe that should trigger an AttributeError... but assuming that does not crash, currentIndex can (in the current shape of things) definitely be set to None. So if not that, we may also want to fix that code while we're about this topic.

Reply:
yeah, that should be `if index and index.isValid()` and the 'if self.currentIndex' is probably older code when it could be set to None.. at least, that's what I'd assume.

Reply:
Agreed. 👍 

Combined with your suggested change, I have removed all checks for `self.currentIndex` and its validity in the signal handlers given that the validity of a current index is already determined when originally setting the index.

Reply:
Well, the non-None-ity of a current index is already determined, but its validity is not. If the current Index refers to the root item, then the index is an invalid QModelIndex as far as I understand it.

Reply:
Right, but I also [read the manual](https://doc.qt.io/qt-5/qmodelindex.html#isValid), which states:

> bool `QModelIndex::isValid()` const
>
> Returns true if this model index is valid; otherwise returns false.
>
> A valid index belongs to a model, and has non-negative row and column numbers.

Row nor column will become negative out of nowhere and it will still be a part of the same model. It does not check upper boundaries. Thus, the check is completely pointless to have, which is why I removed it.

Note that this is official Qt documentation, and not just an implementation detail dug up from a source file. If it were an implementation detail, I would have erred on the side of caution, but it literally describes the conditions the function is supposed to check.

Reply:
Yeah, but the same manual (at the top also says : 
> New QModelIndex objects are created by the model using the QAbstractItemModel::createIndex() function. An invalid model index can be constructed with the QModelIndex constructor. Invalid indexes are often used as parent indexes when referring to top-level items in a model.

In this case, if the editor is set to displaying the "Root", then the currentIndex.isValid() would return False, because to set it to displaying the Root item, we do :

            self.currentIndex = QModelIndex()

And I was replying to your comment that said "I have removed all checks for self.currentIndex and its validity in the signal handlers given that the validity of a current index is already determined when originally setting the index." and my point was that the validity of a current index is not determined because we are originally setting the index to `QModelIndex()` which by definition is invalid. As for whether or not we need to check for the validity in the signal handlers, I'm not sure.

Reply:
Good points, I had not spotted that paragraph.

That said, I tried deleting all folders and texts from my model, adding new texts, removing texts again and repeated a few more times in slightly different ways.. and I do not trigger any problems. Even when the root folder displays when performing these actions I see absolutely no errors thrown. So from testing, the current code seems stable.

(I also managed to open up the root as a text document using the context menu... the editor really needs a thorough overhaul... ☹️ )

Reply:
Yeah, it wouldn't be a problem, you just wouldn't find a common ancestor if you delete something and current index is root. (since root itself can never be deleted) and yes, there are issues like that which might happen with deleting, see #516 for example.

Reply:
Right, but that is something I can't do anything with here. To summarize: the signal handlers, with their checks removed, seem to withstand all the tests I can throw at them involving displaying the root. So while I agree that an invalid QModelIndex could be assigned, it does not seem to be any issue. (Or if it actually is an issue, it is indistinguishable from the myriad other editor oddities we could already observe while messing around in the editor. 😬 )

Based purely on my tests, I say the code is fine like this. If we want those checks put back in, just give a shout and I'll put them back in.

Reply:
I should really not comment on issues when I am tired, because I clearly didn't understand. 😭 Re-reading while awake, you are clearly in agreement that removing these checks isn't a problem.

As for #516, that was an easy fix to implement while chasing up my sleepy misunderstandings, so I'll incorporate it into this PR. And because I am changing this commit anyways, I might as well implement the stupid check and be done with the matter for safeties sake. 😈